clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -246,7 +246,10 @@ public LeaderFollowerStoreIngestionTask(
             .build();
     this.veniceWriter = Lazy.of(() -> veniceWriterFactory.createVeniceWriter(writerOptions));
     this.kafkaClusterIdToUrlMap = serverConfig.getKafkaClusterIdToUrlMap();
-    this.kafkaDataIntegrityValidatorForLeaders = new KafkaDataIntegrityValidator(kafkaVersionTopic);
+    this.kafkaDataIntegrityValidatorForLeaders = new KafkaDataIntegrityValidator(
+        kafkaVersionTopic,
+        KafkaDataIntegrityValidator.DISABLED,
+        getServerConfig().getDivProducerStateMaxAgeMs());
```

Comment:
`kafkaDataIntegrityValidatorForLeaders` should never persist offset, so shall we disable this DIV producer cleanup feature explicitly?
Maybe we should introduce a flag to indicate that `kafkaDataIntegrityValidatorForLeaders` should never call `updateOffsetRecordForPartition`, and if that happens, an exception should be thrown to fail fast.

Reply:
Hmm... we could hardcode it to disabled, but IDK if that is ideal. The goal is not just to clear old state to be persisted, but also to clear old state which is hanging in memory. Actually, the latter is the main motivation for this change, since unbounded memory growth can lead to OOMing, whereas unbounded growth of persistent storage is not a concern (given that metadata is pretty small overall...).

That being said, maybe there is something special about the usage of this "DIV for leaders" which makes it not prone to leaking old state. I'm not sure about that.